cores/epoxy/Arduino.cpp

```diff
@@ -22,6 +22,11 @@
 // -----------------------------------------------------------------------
 
 void yield() {
+  char c = '\0';
+  if (read(STDIN_FILENO, &c, 1) == 1) {
```

Comment:
Thanks for the fix. Just out of curiosity, how did you hit this bug? Sending a NUL character into the Serial port seems like a rare thing.. unless you were sending binary data?

Reply:
I didn't actually run into that bug. I was fixing the unused result compiler warning and noticed that it would also fix the NUL issue.

Reply:
The strange thing is that I have both `-Wall` and `-Wextra` enabled by default. According to the documentations I see online, those imply `-Wunused-result` which should warn me about this. But neither g++ nor clang++ on Linux (Ubuntu 20.04) will trigger a warning for me.

Anyway, I will merge this.